pkg/kubelet/dockertools/labels.go

```diff
@@ -67,6 +70,11 @@ func newLabels(container *api.Container, pod *api.Pod, restartCount int) map[str
 	labels[kubernetesContainerRestartCountLabel] = strconv.Itoa(restartCount)
 	labels[kubernetesContainerTerminationMessagePathLabel] = container.TerminationMessagePath
 
+	// Add annotation for cAdvisor to handle custom metrics if required.
+	if enableCustomMetrics {
+		addCustomMetricsLabelIfRequired(pod, labels)
```

Comment:
I suspect he was just trying to make use of the existing cAdvisor functionality (in which any Docker container can expose metrics as such) without having to change cAdvisor itself.  There is some ongoing discussion about an cAdvisor API for configuring the metrics collection (instead of going through a label and a file), but I'm unsure where that is going at the moment.